resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java

```diff
@@ -1045,12 +1045,12 @@ int dnsRedirectPort(InetAddress server) {
             QuerySucceededEvent succeededEvent = (QuerySucceededEvent) observer.events.poll();
 
             if (cache) {
-                assertNull(nsCache.cache.get("io.", null));
-                assertNull(nsCache.cache.get("netty.io.", null));
+                assertTrue(nsCache.cache.get("io.", null).isEmpty());
```

Comment:
does it ever make sense to return an empty list, as opposed to just null? if we return empty lists then conditionals tend to get more complicated (e.g. `if not empty and not null`), and the tests should likely be updated to reflect this too.

Reply:
@Scottmitch I was thinking about this and decided that it may be better to return an empty list mainly because we not state in the javadocs that we will ever return null from the `DnsCache` interface.

Reply:
We have been returning `null` so perhaps it is best to update the javadocs. `null` is part of the language and there is no reliable way to express a java 6 API which won't return null, which means folks have to `null` check anyways at runtime.